packages/core/scripts/gas-rebate/index.js

```diff
@@ -0,0 +1,301 @@
+          claim: {
+            transactionBlock: transactionBlock.number,
+            hash: transactionReceipt.transactionHash,
+            gasUsed
+          }
+        };
+
+        rewardedVotersToRebate[key] = val;
+        progressBarClaim.update(i + 1);
+      }
+      progressBarClaim.stop();
+      console.log("✅ Finished parsing CLAIM data.");
+
+      // Rebate voters
+      console.log(`${Object.keys(rewardedVotersToRebate).length} Voters Claimed Rewards`);
+      const rebateReceipts = {};
+      let totalGasUsed = 0;
+      let totalEthSpent = 0;
+      let totalUmaRepaid = 0;
+      for (let voterKey of Object.keys(rewardedVotersToRebate)) {
+        const claimData = rewardedVotersToRebate[voterKey].claim;
+        const gasUsed = claimData.gasUsed;
+        const ethToPay = toBN(averagePriceGweiForPeriod).mul(toBN(gasUsed));
+        const umaToPay = ethToPay.mul(ethToUma).div(SCALING_FACTOR);
+        const claimTxn = claimData.hash;
+
+        totalGasUsed += gasUsed;
+        totalEthSpent += Number(fromWei(ethToPay.toString()));
+        totalUmaRepaid += Number(fromWei(umaToPay.toString()));
+
+        rebateReceipts[voterKey] = {
+          gasUsed,
+          ethToPay: Number(fromWei(ethToPay)),
+          umaToPay: Number(fromWei(umaToPay)),
+          claimTxn
+        };
+
+        const voter = rewardedVotersToRebate[voterKey].voter;
+        if (rebateOutput.shareHolderPayout[voter.toLowerCase()]) {
+          rebateOutput.shareHolderPayout[voter.toLowerCase()] += Number(fromWei(umaToPay.toString()));
+        } else {
+          rebateOutput.shareHolderPayout[voter.toLowerCase()] = Number(fromWei(umaToPay.toString()));
+        }
+      }
+
+      console.table(rebateReceipts);
+      console.log(
+        `💎 Prices: {average gas price for period (gwei): ${_averagePriceGweiForPeriod}, average ETH-USD price for period: ${_averageEthPriceForPeriod}, current UMA-USD price: ${_currentUmaPriceForPeriod}}`
+      );
+      console.log(
+        `㊗️ Totals: {gas: ${totalGasUsed.toLocaleString()}, ETH: ${totalEthSpent.toLocaleString()}, UMA: ${totalUmaRepaid.toLocaleString()}}`
+      );
+    }
+
+    // Output JSON parseable via disperse.app
+    let totalUMAToRebate = 0;
+    for (let voter of Object.keys(rebateOutput.shareHolderPayout)) {
+      totalUMAToRebate += rebateOutput.shareHolderPayout[voter];
+    }
+
+    console.log("\n\n*=======================================*");
+    console.log("*                                       *");
+    console.log("* 🧮 Final UMA Rebate                   *");
+    console.log("*                                       *");
+    console.log("*=======================================*");
+    console.log(`🎟 UMA to rebate: ${totalUMAToRebate}`);
+    console.log(`📒 Output JSON: ${JSON.stringify(rebateOutput, null, 4)}`);
+
+    // Format output and save to file.
+    const savePath = `${path.resolve(__dirname)}/weekly-payouts/Week_${weekNumber}_Gas_Rebate.json`;
```

Comment:
nit: is this a `weekly-payout`? I think a better name could be based off the voting rounds prehaps?

Reply:
also you might consider adding additional information to the output such as the voting round, total number of commits, total number of revelas ect. if you'd like.

Reply:
True, I'll just replace Week with Round